src/ralph/backends/storage/swift.py

```diff
@@ -0,0 +1,170 @@
+"""Swift storage backend for Ralph"""
+
+import inspect
+import logging
+import sys
+
+from swiftclient.exceptions import ClientException
+from swiftclient.service import SwiftError, SwiftService, SwiftUploadObject
+
+from ralph.defaults import (
+    SWIFT_OS_AUTH_URL,
+    SWIFT_OS_IDENTITY_API_VERSION,
+    SWIFT_OS_PROJECT_DOMAIN_NAME,
+    SWIFT_OS_USER_DOMAIN_NAME,
+)
+from ralph.exceptions import BackendParameterException
+
+from ..mixins import HistoryMixin
+from .base import BaseStorage
+
+logger = logging.getLogger(__name__)
+
+
+def throws_swift_error(func):
+    """Decorator catches SwiftError or ClientException, then logs and re-raises it"""
+
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except (SwiftError, ClientException) as error:
+            logger.error(error.value)
+            raise error
+
+    return wrapper
+
+
+class SwiftStorage(HistoryMixin, BaseStorage):
+    """OVH's Swift storage backend"""
+
+    name = "swift"
+
+    # pylint: disable=too-many-arguments
+
+    def __init__(
+        self,
+        os_tenant_id,
+        os_tenant_name,
+        os_username,
+        os_password,
+        os_region_name,
+        os_storage_url,
+        os_user_domain_name=SWIFT_OS_USER_DOMAIN_NAME,
+        os_project_domain_name=SWIFT_OS_PROJECT_DOMAIN_NAME,
+        os_auth_url=SWIFT_OS_AUTH_URL,
+        os_identity_api_version=SWIFT_OS_IDENTITY_API_VERSION,
```

Comment:
Here are the most common / necessary options for the swift client.
The swift client support a lot more [options](https://docs.openstack.org/python-swiftclient/latest/service-api.html#options). Do we add (copy/paste) all of them here ?